VultisigApp/VultisigApp/View Models/VaultDetailViewModel.swift

```diff
@@ -17,20 +17,49 @@ class VaultDetailViewModel: ObservableObject {
     func getTotalUpdatedBalance() async {
         var totalBalance: Decimal = 0
         
-        for group in coinsGroupedByChains {
-            for coin in group.coins {
-                do {
-                    let (_, _, balanceInFiatDecimal) = try await BalanceService.shared.balance(for: coin)
-                    totalBalance += balanceInFiatDecimal
-                } catch {
-                    print("Error fetching balance for coin \(coin): \(error)")
+        await withTaskGroup(of: Decimal?.self) { group in
+            for groupedChain in coinsGroupedByChains {
+                for coin in groupedChain.coins {
+                    group.addTask {
+                        do {
+                            return try await self.fetchBalanceWithRetry(for: coin, retries: 3, delay: 1.0)
+                        } catch {
+                            print("Error fetching balance for coin \(coin): \(error)")
+                            return nil
+                        }
+                    }
+                }
+            }
+            
+            for await result in group {
+                if let balance = result {
+                    totalBalance += balance
                 }
             }
         }
         
         self.totalBalanceInFiat = totalBalance
     }
     
+    private func fetchBalanceWithRetry(for coin: Coin, retries: Int, delay: TimeInterval) async throws -> Decimal {
```

Comment:
since you already retry in the service , like RpcService and PolkadotService , so here you don't need to retry , otherwise it is trying 3*3 times

Reply:
This is also important, because we have Solana, Sui, Thor, Maia, Gaia, Kujira, and others. It will help us when calculating the Super Total. It is also in parallel. So in case we have 10 coins it will start multiple tasks.

Reply:
No, what I mean is , if the RPC call to fetch balance is already with retry , then here , we don't need to retry it again , otherwise it is getting multiple retries